### PR TITLE
Adding a new VMware utilities module

### PR DESCRIPTION
**Pull request by jcpowermac**
This PR adds the support for VMware utility module for frequently used functions such as:

`wait_for_task`
`find_datacenter_by_name`
`vmware_argument_spec`
`connect_to_api`
`get_all_objs`

This [PR](https://github.com/ansible/ansible-modules-core/pull/1154) for the cluster and datacenter is using this module and has been tested.

### CODE REVIEW

File: lib/ansible/module_utils/vmware.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
+#!/bin/python
```
Comment (bcoca) [inline lib/ansible/module_utils/vmware.py]: this is for import, should not have a shebang

File: lib/ansible/module_utils/vmware.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+def find_dvs_by_name(content, switch_name):
+    try:
+        vmware_distributed_switches = get_all_objs(content, [vim.dvs.VmwareDistributedVirtualSwitch])
+        for dvs in vmware_distributed_switches:
+            if dvs.name == switch_name:
+                return dvs
+        return None
+    # This exception should be handled in the module that calls this method
+    # and fail with an appropriate message to module.fail_json()
+    except vmodl.MethodFault:
```
Comment (bcoca) [inline lib/ansible/module_utils/vmware.py]: useless try/except, several of these in the code

File: lib/ansible/module_utils/vmware.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        # Such as IP change of the ESXi host which removes the connection anyway.
+        # Also removal significantly speeds up the return of the module
+
+        if disconnect_atexit:
+            atexit.register(connect.Disconnect, service_instance)
+        return service_instance.RetrieveContent()
+    except vim.fault.InvalidLogin as invalid_login:
+        module.fail_json(msg=invalid_login.msg)
+    except requests.ConnectionError:
+        module.fail_json(msg="Unable to connect to vCenter or ESXi API on TCP/443.")
```
Comment (bcoca) [inline lib/ansible/module_utils/vmware.py]: it is useful to also add the api exception info:
module.fail_json(msg="Unable to connect to vCenter or ESXi API on TCP/443.", apierror=str(e))

File: lib/ansible/module_utils/vmware.py [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+def task_success(task):
+    return True
+
+
+def task_running(task):
+    time.sleep(15)
+    return False
+
+
+def task_error(task):
```
Comment (bcoca) [inline lib/ansible/module_utils/vmware.py]: I'm not sure this function is useful

File: lib/ansible/module_utils/vmware.py [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+class TaskError(Exception):
+    pass
+
+
+def task_success(task):
+    return True
+
+
+def task_running(task):
+    time.sleep(15)
```
Comment (bcoca) [inline lib/ansible/module_utils/vmware.py]: task doesn't seem to be used here

File: lib/ansible/module_utils/vmware.py [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+except ImportError:
+    HAS_PYVMOMI = False
+
+
+class TaskError(Exception):
+    pass
+
+
+def task_success(task):
+    return True
```
Comment (bcoca) [inline lib/ansible/module_utils/vmware.py]: not sure if this is useful

**Comment by jcpowermac** [comment]
The replacement module-extras [PR](https://github.com/ansible/ansible-modules-extras/pull/415) specifically for adding datacenter objects within vCenter, which uses this utility module.

File: lib/ansible/module_utils/vmware.py [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+def find_cluster_by_name_datacenter(datacenter, cluster_name):
+
+    host_folder = datacenter.hostFolder
+    for folder in host_folder.childEntity:
+        if folder.name == cluster_name:
+            return folder
+    return None
+
+
+def find_datacenter_by_name(content, datacenter_name, throw=True):
```
Comment (bcoca) [inline lib/ansible/module_utils/vmware.py:73]: throw seems unused

**Pull request merged**